### PR TITLE
Simplified sample [API-1977]

### PR DESCRIPTION
**Pull request by emreyigit**

### CODE REVIEW

**Comment by emreyigit** [comment]
cc: @OzanCansel @akeles85

File: client_with_ssl.cpp [hunk 1]
```diff
@@ -1,84 +1,176 @@
...
+	client.shutdown().get();
 }
 
-hazelcast::client::hazelcast_json_value as_json(std::string country, std::string city, int population) {
-    return hazelcast::client::hazelcast_json_value("{\"country\":\"" + country + "\",\"city\":\"" + city + "\",\"population\":" +std::to_string(population)  +"}");
+void create_mapping(hazelcast::client::hazelcast_client client) {
+	// Mapping is required for your distributed map to be queried over SQL.
+	// See: https://docs.hazelcast.com/hazelcast/latest/sql/mapping-to-maps
+
+	std::count << "\nCreating the mapping...";
```
Comment (akeles85) [inline client_with_ssl.cpp:63]:
```suggestion
	std::cout << "\nCreating the mapping..." << std::endl;
```

File: client_with_ssl.cpp [hunk 2]
```diff
@@ -1,84 +1,176 @@
...
-* <p>
-* See: <a href="https://docs.hazelcast.com/cloud/cpp-client">https://docs.hazelcast.com/cloud/cpp-client</a>
+* A sample application that configures a client to connect to an Hazelcast Viridian cluster
+* over TLS, and to then insert and fetch data with SQL, thus testing that the connection to
+* the Hazelcast Viridian cluster is successful.
+*
+* See: https://docs.hazelcast.com/cloud/get-started
 */
 
 void map_example(hazelcast::client::hazelcast_client client);
```
Comment (akeles85) [inline client_with_ssl.cpp:26]: these function declarations should be deleted;
void map_example(hazelcast::client::hazelcast_client client);
void non_stop_map_example(hazelcast::client::hazelcast_client client);

These function declarations should be added.
void create_mapping(hazelcast::client::hazelcast_client client);
void insert_cities(hazelcast::client::hazelcast_client client);

File: client_with_ssl.cpp [hunk 3]
```diff
@@ -1,84 +1,176 @@
...
+                                        __key INT,                                        
+                                        country VARCHAR,
+                                        city VARCHAR,
+                                        population INT) TYPE IMAP
+                                    OPTIONS ( 
+                                        'keyFormat' = 'int',
+                                        'valueFormat' = 'compact',
+                                        'valueCompactTypeName' = 'city'))").get();
+
+	std::count << "OK.";
```
Comment (akeles85) [inline client_with_ssl.cpp:79]:
```suggestion
	std::cout << "OK." << std::endl;
```

File: client_with_ssl.cpp [hunk 4]
```diff
@@ -1,84 +1,176 @@
...
+                    (__key, city, country, population) VALUES
+                    (1, 'London', 'United Kingdom', 9540576),
+                    (2, 'Manchester', 'United Kingdom', 2770434),
+                    (3, 'New York', 'United States', 19223191),
+                    (4, 'Los Angeles', 'United States', 3985520),
+                    (5, 'Istanbul', 'T�rkiye', 15636243),
+                    (6, 'Ankara', 'T�rkiye', 5309690),
+                    (7, 'Sao Paulo ', 'Brazil', 22429800))").get();
+
+	std::count << "OK.";
```
Comment (akeles85) [inline client_with_ssl.cpp:99]:
```suggestion
	std::cout << "OK." << std::endl;
```

File: client_with_ssl.cpp [hunk 5]
```diff
@@ -1,84 +1,176 @@
...
+
+	// Let's also add a city as object.
+	std::cout << std::endl << "Putting a city into 'cities' map...";
+	auto map = client.get_map("cities").get();
+	map->put(8, CityDTO{ "Rio de Janeiro", "Brazil", 13634274 })
```
Comment (akeles85) [inline client_with_ssl.cpp:104]:
```suggestion
	map->put(8, CityDTO{ "Rio de Janeiro", "Brazil", 13634274 });
```

File: client_with_ssl.cpp [hunk 6]
```diff
@@ -1,84 +1,176 @@
...
+		std::cout << "OK.";
```
Comment (akeles85) [inline client_with_ssl.cpp:105]:
```suggestion
	std::cout << "OK." << std::endl;
```

File: client_with_ssl.cpp [hunk 7]
```diff
@@ -1,84 +1,176 @@
...
+			auto id = row.get_object<int32_t>("__key");
+			auto city = row.get_object<CityDTO>("this");
+			std::printf("| %4d | %20s | %20s | %15d |\n", id, city.Country, city.cityName, city.population);			
+		}
+	}
+
+	std::cout << "\n!! Hint !! You can execute your SQL queries on your Viridian cluster over the management center. \n 1. Go to 'Management Center' of your Hazelcast Viridian cluster. \n 2. Open the 'SQL Browser'. \n 3. Try to execute 'SELECT * FROM cities'.\n";
 }
+struct CityDTO {
+	string cityName;
```
Comment (akeles85) [inline client_with_ssl.cpp:131]:
```suggestion
	std::string cityName;
```

File: client_with_ssl.cpp [hunk 8]
```diff
@@ -1,84 +1,176 @@
...
+	std::printf("| %4s | %20s | %20s | %15s |\n", "id", "country", "city", "population");
+
+	for (auto itr = result->iterator(); itr.has_next();) {
+		auto page = itr.next().get();
+
+		for (auto const& row : page->rows()) {
+
+			auto id = row.get_object<int32_t>("__key");
+			auto city = row.get_object<CityDTO>("this");
+			std::printf("| %4d | %20s | %20s | %15d |\n", id, city.Country, city.cityName, city.population);
```
Comment (akeles85) [inline client_with_ssl.cpp:124]:
```suggestion
			std::printf("| %4d | %20s | %20s | %15d |\n", id, city.country, city.cityName, city.population);			
```

**Review by OzanCansel — APPROVED** [review]

**Review by akeles85 — APPROVED** [review]
I have made the necessary changes and push them according to my comments.

**Comment by akeles85** [comment]
We have tested the example for windows, linux and macos.

As we see there is some error with documentation. We should add SSL_CERT_FILE as enviroment variable with key like "C:\cacert.pem". 

Installation should be changed:
vcpkg install "hazelcast-cpp-client[openssl]:x64-windows"

We should also change the documentation for starting the example like below:

cmake -B build -S . -DCMAKE_TOOLCHAIN_FILE=[path to vcpkg]\scripts\buildsystems\vcpkg.cmake && ^
cmake --build build && ^
build\Debug\client_with_ssl.exe

File: client_with_ssl.cpp [hunk 9]
```diff
@@ -1,84 +1,224 @@
...
+          });
+        ctx.use_private_key_file("key.pem", boost::asio::ssl::context::pem);
+    } catch (std::exception& e) {
+        std::cerr << "You should copy ca.pem, cert.pem and key.pem files to "
+                     "the working directory, exception cause "
+                  << e.what() << std::endl;
+        exit(EXIT_FAILURE);
+    }
     config.get_network_config().get_ssl_config().set_context(std::move(ctx));
+    config.get_logger_config().level(hazelcast::logger::level::off);
```
Comment (ihsandemir) [inline client_with_ssl.cpp]: why?

**Review by ihsandemir — CHANGES_REQUESTED** [review]

File: client_with_ssl.cpp (hunk 9 above)
Comment (emreyigit) [reply client_with_ssl.cpp]: fixed at https://github.com/hazelcast/hazelcast-cloud-cpp-sample-client/pull/10/commits/a055c5e5b80c1df85ab07503c2ef208d66a3a643

Comment (akeles85) [reply client_with_ssl.cpp]: logs seems ugly when the example runs. So I made them off. Other clients also do not show logs.

**Review by abramche — APPROVED** [review]

File: client_with_ssl.cpp (hunk 9 above)
Comment (abramche) [reply client_with_ssl.cpp]: @ihsandemir all good by you? I can merge the changes after you resolve the review :)

**Review by ihsandemir — APPROVED** [review]

File: client_with_ssl.cpp (hunk 9 above)
Comment (ihsandemir) [reply client_with_ssl.cpp]: approved @EugeneAbramchuk

**Review by abramche — APPROVED** [review]

**Pull request merged**